core/src/main/java/org/elasticsearch/index/query/MatchQueryBuilder.java

```diff
@@ -60,8 +60,10 @@
     public static final ParseField TYPE_FIELD = new ParseField("type");
     public static final ParseField QUERY_FIELD = new ParseField("query");
 
-    /** The default name for the match query */
-    public static final String NAME = "match";
+    /** The names for the match query. */
+    // TODO replace this with a Types like object
+    public static final Names NAMES = new Names("match", "match_phrase", "matchPhrase", "match_phrase_prefix", "matchPhrasePrefix",
```

Comment:
I see where these concerns come from. Moving to ParseField is a bigger change... cause it would require to split the match query into multiple queries: match, match_phrase, match_phrase_prefix, match_fuzzy... then all the other options are other camelCase or deprecated. Although this sounds good conceptually, I am not sure it will play well with current code. Let's not do that in this PR them which means that we'll have to keep Names, at least temporarily?